drivers/wifi/infineon/airoc_wifi.c

```diff
@@ -0,0 +1,652 @@
+			return;
+		}
+
+		pkt = net_pkt_rx_alloc_with_buffer(airoc_wifi_iface, len, AF_UNSPEC, 0, K_NO_WAIT);
+
+		if (pkt != NULL) {
+			if (net_pkt_write(pkt, data, len) < 0) {
+				LOG_ERR("Failed to write pkt");
+				net_pkt_unref_flag = true;
+			}
+
+			if ((net_pkt_unref_flag) || (net_recv_data(airoc_wifi_iface, pkt) < 0)) {
+				LOG_ERR("Failed to push received data");
+				net_pkt_unref_flag = true;
+			}
+		} else {
+			LOG_ERR("Failed to get net buffer");
+		}
+	}
+	/* Release a packet buffer */
+	cy_buffer_release(buffer, WHD_NETWORK_RX);
+
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+	airoc_wifi_data.stats.bytes.received += len;
+	airoc_wifi_data.stats.pkts.rx++;
+#endif
+
+	if (net_pkt_unref_flag) {
+		net_pkt_unref(pkt);
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+		airoc_wifi_data.stats.errors.rx++;
+#endif
+	}
+}
+
+static void *link_events_handler(whd_interface_t ifp, const whd_event_header_t *event_header,
+				 const uint8_t *event_data, void *handler_user_data)
+{
+	ARG_UNUSED(ifp);
+	ARG_UNUSED(event_data);
+	ARG_UNUSED(handler_user_data);
+
+	(void)k_msgq_put(&airoc_wifi_msgq, event_header, K_FOREVER);
+	return NULL;
+}
+
+static void airoc_event_task(void)
+{
+	whd_event_header_t event_header;
+
+	while (1) {
+		(void)k_msgq_get(&airoc_wifi_msgq, &event_header, K_FOREVER);
+
+		switch ((whd_event_num_t)event_header.event_type) {
+		case WLC_E_LINK:
+			break;
+
+		case WLC_E_DEAUTH_IND:
+		case WLC_E_DISASSOC_IND:
+			net_if_dormant_on(airoc_wifi_iface);
+			break;
+
+		default:
+			break;
+		}
+	}
+}
+
+static void airoc_mgmt_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct airoc_wifi_data *data = dev->data;
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+
+	eth_ctx->eth_if_type = L2_ETH_IF_TYPE_WIFI;
+	data->iface = iface;
+	airoc_wifi_iface = iface;
+
+	/* Read WLAN MAC Address */
+	(void)whd_wifi_get_mac_address(airoc_if, &airoc_if->mac_addr);
+	(void)memcpy(&data->mac_addr, &airoc_if->mac_addr, sizeof(airoc_if->mac_addr));
+
+	/* Assign link local address. */
+	(void)net_if_set_link_addr(iface, data->mac_addr, 6, NET_LINK_ETHERNET);
+
+	(void)ethernet_init(iface);
+
+	/* Not currently connected to a network */
+	(void)net_if_dormant_on(iface);
+
+	/* L1 network layer (physical layer) is up */
+	(void)net_if_carrier_on(data->iface);
+}
+
+whd_result_t cy_buffer_pool_init(void *tx_packet_pool, void *rx_packet_pool)
+{
+	CY_UNUSED_PARAMETER(tx_packet_pool);
+	CY_UNUSED_PARAMETER(rx_packet_pool);
+
+	return 0;
+}
+
+static void scan_cb_search(whd_scan_result_t **result_ptr, void *user_data,
+			   whd_scan_status_t status)
+{
+	if (status == WHD_SCAN_ABORTED) {
+		k_sem_give(&airoc_wifi_data.sema_scan);
+		return;
+	}
+	if (status == WHD_SCAN_COMPLETED_SUCCESSFULLY) {
+		k_sem_give(&airoc_wifi_data.sema_scan);
+	} else if (status == WHD_SCAN_INCOMPLETE) {
+		if (user_data != NULL) {
+			if ((**result_ptr).SSID.length > 0) {
+				if (strcmp((**result_ptr).SSID.value,
```

Comment:
In general length based string APIs are safe, in this case `strncmp` unless you know both are NULL terminated.

Reply:
updated